src/main/java/org/jetbrains/bsp/bazel/server/BepServer.java

```diff
@@ -88,156 +86,146 @@ public void publishLifecycleEvent(
   @Override
   public StreamObserver<PublishBuildToolEventStreamRequest> publishBuildToolEventStream(
       StreamObserver<PublishBuildToolEventStreamResponse> responseObserver) {
-    return new StreamObserver<PublishBuildToolEventStreamRequest>() {
-
-      @Override
-      public void onNext(PublishBuildToolEventStreamRequest request) {
-        if (request
-            .getOrderedBuildEvent()
-            .getEvent()
-            .getBazelEvent()
-            .getTypeUrl()
-            .equals("type.googleapis.com/build_event_stream.BuildEvent")) {
-          handleEvent(request.getOrderedBuildEvent().getEvent());
-        } else {
-          //          System.out.println("Got this request " + request);
-        }
-        PublishBuildToolEventStreamResponse response =
-            PublishBuildToolEventStreamResponse.newBuilder()
-                .setStreamId(request.getOrderedBuildEvent().getStreamId())
-                .setSequenceNumber(request.getOrderedBuildEvent().getSequenceNumber())
-                .build();
-        responseObserver.onNext(response);
-      }
+    return new BepStreamObserver(this, responseObserver);
+  }
 
-      private void handleEvent(BuildEvent buildEvent) {
-        try {
-          BuildEventStreamProtos.BuildEvent event =
-              BuildEventStreamProtos.BuildEvent.parseFrom(buildEvent.getBazelEvent().getValue());
-          System.out.println("Got event" + event + "\nevent-end\n");
-          if (event.hasStarted() && event.getStarted().getCommand().equals("build")) {
-            BuildEventStreamProtos.BuildStarted buildStarted = event.getStarted();
-            TaskId taskId = new TaskId(buildStarted.getUuid());
-            TaskStartParams startParams = new TaskStartParams(taskId);
-            startParams.setEventTime(buildStarted.getStartTimeMillis());
-            bspClient.onBuildTaskStart(startParams);
-            taskParkingLot.add(taskId);
-          }
-          if (event.hasFinished()) {
-            BuildEventStreamProtos.BuildFinished buildFinished = event.getFinished();
-            if (taskParkingLot.size() == 0) {
-              System.out.println("No start event id was found.");
-              return;
-            } else if (taskParkingLot.size() > 1) {
-              System.out.println("More than 1 start event was found");
-              return;
-            }
+  public void collectDiagnostics(
+      Map<Uri, List<PublishDiagnosticsParams>> filesToDiagnostics,
+      BuildTargetIdentifier target,
+      String diagnosticsLocation)
+      throws IOException {
```

Comment:
Ideally we'd like to get rid of checked exceptions, so you can wrap this error *inside* `diagnosticsDispatcher`